src/samples.jl

```diff
@@ -371,13 +371,15 @@ If:
 
     sample_data isa AbstractArray &&
     sample_resolution_in_unit == 1 &&
-    sample_offset_in_unit == 0
+    sample_offset_in_unit == 0 &&
+    eltype(sample_data) == typeof(sample_resolution_in_unit) == typeof(sample_offset_in_unit)
```

Comment:
when called with `Samples`, `sample_resolution_in_unit` and `sample_offset_in_unit` will _always_ be Float64, so this effectively means decode will _always_ return `Float64` values.  this is bad for both "categorical" type signals (where the signal data actually _is_ an integer which represent an index) and for `Float32` signals which a lot of ML models use.

Reply:
The reason this guard is here in teh first place is that it _used_ to be possible (in Onda 0.14 I think) to use integer 0/1 for the offset/resolution in `SamplesInfo`: https://github.com/beacon-biosignals/Onda.jl/blob/2e33a6e33dedb355d6ac7400f98f43adf6878f7b/src/signals.jl#L66-L67

That changed with the switch to `SamplesInfoV2`: https://github.com/beacon-biosignals/Onda.jl/blob/36a7be9080b13aa670445922e451cabd0c2673be/src/signals.jl#L80-L81

Reply:
Are categorical signals encoded at all? It seems like they should use `encoded=false` which would address that concern. I agree that `Float32` shouldn't be converted to a `Float64`.

Reply:
`encoded` is not a property of the stored signal, it's specific to in-memory `Samples` objects, so there's no way for a signal to be encoded or decoded.

Reply:
That's definitely annoying...